fr-FR: Improve translation

rmnvgr opened the pull request:
- All quotation marks replaced with guillemets
- Text that should be in italics put in small fonts
- Correct spaces around punctuation
- Correct capitalization
- Accents on uppercase characters
- Correct apostrophes
- Correct dashes
- Follow [translation guidelines](https://github.com/OpenRCT2/Localisation/wiki/Translator-guidelines:-fr_FR)

Gymnasiast commented on data/language/fr-FR.txt:
```diff
@@ -866,131 +866,131 @@ STR_1476    :{WINDOW_COLOUR_2}Niveau d’intensité : {BLACK}Pas encore disponi
-STR_1525    :“Ces frites de {STRINGID} sont un très bon choix”
-STR_1526    :“Cette glace de {STRINGID} est un très bon choix”
-STR_1527    :“Cette barbe-à-papa de {STRINGID} est un très bon choix”
+STR_1480    :« Je ne peux pas me payer {STRINGID} »
```
Are these regular spaces? It’s better to use non-breaking spaces, so you won't have «
situations like these ».

rmnvgr replied:
They are non-breaking spaces.

Gymnasiast commented on data/language/fr-FR.txt:
```diff
@@ -1828,8 +1828,8 @@ STR_2481    :Voir graphique Accélération latérale sur durée
 STR_2484    :{WINDOW_COLOUR_2}Bénéfice hebdomadaire : {RED}{CURRENCY2DP}
-STR_2487    :Afficher le ‘vrai’ nom des visiteurs
-STR_2488    :Passer du ‘vrai’ nom des visiteurs à leur numéro
+STR_2487    :Afficher le « vrai » nom des visiteurs
```
This is not a quotation. Any reason why you gave this (and several other strings) guillemets, while other strings (e.g. 2742) got {SMALLFONT}?

rmnvgr replied:
Yes, when a word should not be taken literally, you put guillemets around it.

rmnvgr commented:
I've added [fr-FR translation guidelines](https://github.com/OpenRCT2/Localisation/wiki/Translator-guidelines:-fr_FR), and updated the translation to follow them.

Gymnasiast commented:
Nice work! Sorry for taking so long to review it again.

Some things:
- You used a longer dash in some cases. This is fine if it is correct for French, but have you checked if OpenRCT2 displays it correctly? If not, I need to add support.
- The translation appears to use both `œ` and `oe`, inconsistently. AFAIK, `œ` is still proper French, is that correct? If so, I’d need to add a glyph for it, as we don’t currently have one.
- Some big changes are afoot for the objects translation, and I’d need to redump them. Since you didn’t yet do much to that (just easy substitutions), could you revert those changes for now? (Probably best to keep those changes somewhere just for reference)
- Could you rebase your pull request?

rmnvgr commented:
> Nice work! Sorry for taking so long to review it again.

No worries, it's quite a substantial PR.

> You used a longer dash in some cases. This is fine if it is correct for French, but have you checked if OpenRCT2 displays it correctly? If not, I need to add support.

Indeed, I didn't check, and OpenRCT2 doesn't display it correctly.

> The translation appears to use both `œ` and `oe`, inconsistently. AFAIK, `œ` is still proper French, is that correct? If so, I’d need to add a glyph for it, as we don’t currently have one.

Good catch. `œ` is correct, I've changed the strings.

> Some big changes are afoot for the objects translation, and I’d need to redump them. Since you didn’t yet do much to that (just easy substitutions), could you revert those changes for now? (Probably best to keep those changes somewhere just for reference)

I've moved the changes to objects to [this branch](https://github.com/rmnvgr/Localisation/tree/improved-objects).

> Could you rebase your pull request?

Done!

Gymnasiast commented on data/language/fr-FR.txt:
```diff
@@ -714,25 +714,25 @@ STR_1324    :{COMMA16} parcours
 STR_1340    :{WINDOW_COLOUR_2}Vitesse maximale : {BLACK}{VELOCITY}
 STR_1341    :{WINDOW_COLOUR_2}Durée de l’attraction : {BLACK}{STRINGID}{STRINGID}{STRINGID}{STRINGID}
 STR_1342    :{DURATION}
-STR_1343    :{DURATION} / 
```
This string and 1346 need the trailing whitespace. It’s not ideal, I know.

Gymnasiast commented on data/language/fr-FR.txt:
```diff
@@ -1116,7 +1116,7 @@ STR_1727    :Les droits de construction ne sont pas à vendre !
 STR_1729    :Ce terrain n’appartient pas au parc !
 STR_1730    :{RED}Fermé
-STR_1731    :{WHITE}{STRINGID} - - 
+STR_1731    :{WHITE}{STRINGID} - -
```
This string and 1778 need the trailing whitespace.

Gymnasiast commented on data/language/fr-FR.txt:
```diff
@@ -2197,20 +2197,20 @@ STR_3063    :Tunnel aquatique (voie engloutie)
-STR_3090    :Choisir style d’entrée, de sortie et de station
-STR_3091    :Interdit de supprimer cette section !
-STR_3092    :Interdit de déplacer ou modifier la station de cette attraction !
+STR_3090    :Choisit  le style d’entrée, de sortie et de station
```
Double space.

```suggestion
STR_3090    :Choisit le style d’entrée, de sortie et de station
```

rmnvgr replied:
Good catch, it is fixed.

Gymnasiast commented on data/language/fr-FR.txt:
```diff
@@ -2354,26 +2354,26 @@ STR_3253    :Impossible de diminuer davantage l’emprunt maximum !
-STR_3259    :Choisir si vous souhaitez que les visiteurs préfèrent des attr. plus intenses
-STR_3260    :{WINDOW_COLOUR_2}Argent par visiteur (moy.) :
+STR_3259    :Les visiteurs préfèrent des attractions plus intenses
+STR_3260    :{WINDOW_COLOUR_2}Argent par visiteur (en moyenne) :
```
This string might be too long for the window in question. Could you check it? (It’s the window in the Finances step of the Scenario Editor.)

rmnvgr replied:
It fits in the window.

Gymnasiast commented on data/language/fr-FR.txt:
```diff
@@ -2493,30 +2493,30 @@ STR_5125    :Tout destructible
 STR_5150    :Activer les outils de débogage
 #Thousands separator
-STR_5151    : 
+STR_5151    :
```
This line needs to be reverted. (See the comment on the preceding line.)

Gymnasiast commented on data/language/fr-FR.txt:
```diff
@@ -2791,7 +2791,7 @@ STR_5463    :But : Amusez-vous !
 STR_5465    :Météo
 STR_5466    :Personnel
-STR_5467    :ALT + 
+STR_5467    :ALT +
```
revert

Gymnasiast commented on data/language/fr-FR.txt:
```diff
@@ -2945,16 +2945,16 @@ STR_5621    :Loopings en Folie
 STR_5632    :Construisez votre propre parc…
-STR_5633    :CMD + 
-STR_5634    :OPTION + 
+STR_5633    :CMD +
```
revert (and next line too)

Gymnasiast commented on data/language/fr-FR.txt:
```diff
@@ -3534,30 +3534,30 @@ STR_6272    :Quais
 STR_6282    :Chat
-STR_6283    :Le chat n’est pas disponible actuellement. Etes-vous connecté(e) à un serveur ?
+STR_6283    :Le chat n’est pas disponible actuellement. Êtes-vous connecté(e) à un serveur ?
 STR_6293    :B
```
I thought French used `o` (octet) instead of `B` (bytes). Or did it switch to `B`?

rmnvgr replied:
No, we use `o`, it just hadn't been translated.

rmnvgr commented:
Sorry for the missing trailing spaces, my editor removes them automatically.

Gymnasiast commented:
Good!

I have now added support for drawing the dashes and I also created sprites for the Œ ligature: https://github.com/OpenRCT2/OpenRCT2/pull/15034

rmnvgr commented:
Nice. I kept the long dashes, I suppose a sprite is also needed for that?

Gymnasiast commented:
> Nice. I kept the long dashes, I suppose a sprite is also needed for that?

I just added code to draw them as regular dashes. If needed/desired, we can add proper glyphs later on.

rmnvgr commented:
Sounds good! 👍
